dcos_test_utils/package.py

```diff
@@ -93,3 +93,206 @@ def list_packages(self):
+
+        :param name: Repository name
+        :type name: str
+        :param uri: Repository URI
+        :type uri: str
+        :return: JSON response
+        :rtype: dict
+        """
+        params = {}
+        if name:
+            params['name'] = name
+        if uri:
+            params['uri'] = uri
+
+        self._update_headers('repository.delete')
+        r = self._post('/delete', params)
+        return r.json()
+
+    def list(self) -> dict:
+        """Get list of package repositories.
+
+        :return: JSON response
+        :rtype: dict
+        """
+        self._update_headers('repository.list')
+        r = self._post('/list', {})
+        return r.json()
+
+
+class Package(Cosmos):
+    def __init__(self, default_url, session=None):
+        super().__init__(default_url, session=session)
+        self._versions = {
+            'request_version':  '1',
+            'response_version': '1',
+        }
+
+    def _make_request(self, resource: str, params: dict,
+                      versions: dict = None) -> dict:
+        """Every request to /packaging has the same couple of steps.
+        This puts them in one place instead of duplicating efforts.
+
+        :param versions: Request and Response versions
+        :type versions: dict
+        :param resource: Package resource (list, install, etc)
+        :type resource: str
+        :param params: POST parameters for the resource
+        :type params: dict
+        :return: JSON response
+        :rtype: dict
+        """
+        kw = versions if versions else self._versions
+        self._update_headers(resource, **kw)
+        r = self._post("/{}".format(resource), params)
+        return r.json()
+
+    def list(self, name: str = None, app_id: str = None) -> dict:
+        """List installed packages.
+
+        :param app_id: App Id (optional)
+        :type app_id: str
+        :param name: Package name (optional)
+        :type name: str
+        :return: installed packages
+        :rtype: dict
+        """
+        params = {}
+        if app_id:
+            params['appId'] = app_id
+        if name:
+            params['packageName'] = name
+        return self._make_request('list', params)
+
+    def install(self, name: str, version: str = None,
+                options: dict = None, app_id: str = None) -> dict:
+        """Install a Universe package.
+
+        :param name: Package name
+        :type name: str
+        :param version: Package version string (optional)
+        :type version: str
+        :param options: Package installation options (optional)
+        :type options: dict
+        :param app_id: App ID (optional)
+        :type app_id: str
+        :return: JSON response
+        :rtype: dict
+        """
+        params = {
+            'packageName': name,
+        }
+        if version is not None:
+            params['packageVersion'] = version
+        if options and type(options) == dict:
```

Comment:
Would it be better to raise a `TypeError` on these instead of silently skipping the param if it's not the expected type?